tests/test_e2e.py

```diff
@@ -156,36 +160,37 @@ def test_greedy(self):
         self.check_accuracy(mapping)
 
 
-class TestEntityMatchTopK(unittest.TestCase):
-    def test_cffi_k(self):
-        nl = randomnames.NameList(300)
-        s1, s2 = nl.generate_subsets(150, 0.8)
+class TestEntityMatchTopK(EntityHelperMixin, unittest.TestCase):
+
+    proportion = 0.8
+    sample = 150
+
+    def setUp(self):
+        self.nl = randomnames.NameList(300)
+        self.s1, self.s2 = self.nl.generate_subsets(self.sample, self.proportion)
         keys = ('test1', 'test2')
-        key_lists = generate_key_lists(keys, len(nl.schema_types))
-        f1 = tuple(bloomfilter.stream_bloom_filters(s1, key_lists, nl.SCHEMA))
-        f2 = tuple(bloomfilter.stream_bloom_filters(s2, key_lists, nl.SCHEMA))
+        self.key_lists = generate_key_lists(keys, len(self.nl.schema_types))
+
+    def test_cffi_k(self):
+
+        f1 = tuple(bloomfilter.stream_bloom_filters(self.s1, self.key_lists, self.nl.SCHEMA))
+        f2 = tuple(bloomfilter.stream_bloom_filters(self.s2, self.key_lists, self.nl.SCHEMA))
 
         threshold = 0.8
```

Comment:
Yes I think this is desired in the end to end test - we want some false positives to be ignored by the next step (the network_flow solver).

Reply:
but since you generate different datasets every test, you cannot know in advance how many matches are going to be above 0.8. Thus your test will either fail every now and then, or the value for allowed false positives is set so high, that the utility of the tests will be fairly limited.

Maybe you want to fix the seed?